server/src/main/java/com/google/coffeehouse/servlets/LeaveClubServlet.java

```diff
@@ -0,0 +1,118 @@
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to leave a club from Http POST Request Body (in JSON format) containing a club ID
+ * and user ID that exists in the database, and return a success status 200 (OK).
+ */
+@WebServlet("/api/leave-club")
+public class LeaveClubServlet extends HttpServlet {
+
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request cannot be used to leave a club for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+  /** The message to be logged when the body of the POST request cannot be parsed. */
+  public static final String LOG_BODY_ERROR_MESSAGE = 
+      "Body unable to be parsed in LeaveClubServlet: ";
+
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request does not have a required field.
+   */
+  public static final String NO_FIELD_ERROR = "Missing \"%s\" field in JSON";
+  /** The message to be logged when the body of the GET request does not have required fields. */
+  public static final String LOG_INPUT_ERROR_MESSAGE =
+      "Error with JSON input in GetProfileServlet: ";
+  /** Name of the key in the input JSON that corresponds to the userId. */
+  public static final String USER_ID_FIELD_NAME = "userId";
+  /** Name of the key in the input JSON that corresponds to the clubId. */
+  public static final String CLUB_ID_FIELD_NAME = "clubId";
+
+  private static final Gson gson = new Gson();
+  private final StorageHandlerApi storageHandler;
+
+  /**
+   * Overloaded constructor for dependency injection.
+   * @param storageHandler the {@link StorageHandlerApi} that is used when deleting the membership
+   */
+  public LeaveClubServlet(StorageHandlerApi storageHandler) {
+    super();
+    this.storageHandler = storageHandler;
+  }
+
+  /**
+   * Explicit default constructor used for instantiating the servlet when not testing.
+   */
+  public LeaveClubServlet() {
+    super();
+    this.storageHandler = new StorageHandlerApi();
+  }
+
+  /** 
+   * Deletes a membership from the database using user ID and club ID.
+   * @param request the POST request that must have a valid JSON representation of the userId and
+   *     clubId to be passed in order to delete a membership from the Memberships table in the
+   *     database. If this is not the case the response will send a "400 Bad Request error".
+   * @param response the response from this method, will set the status to 200 (OK).
+   *     If the request object does not have a valid JSON body representation of the required
+   *     fields userId and clubId, the response will send a "400 Bad Request error". If the request
+   *     object is attempting to delete a membership which does not exist in the database, the
+   *     response will send a "409 Conflict" status. If the request object is unable to be parsed,
+   *     the response will send a "500 Internal Server error".
```

Comment:
It seems the doc comment for `@param request` disagrees with the doc comment for `@param response` on what error codes will be sent. Also it's standard to not end @param doc comments with a period.

Reply:
I'll make sure to change the `@param request` comment. Thanks for point this out!